lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -170,109 +177,121 @@ def install_libraries
               pod_target.umbrella_header_path
             end.compact.group_by(&:dirname)
 
-            pod_targets.sort_by(&:name).each do |pod_target|
-              target_installer = PodTargetInstaller.new(sandbox, project, pod_target, umbrella_headers_by_dir)
-              target_installer.install!
-            end
+            pod_target_installation_results = Hash[pod_targets.sort_by(&:name).map do |pod_target|
+              target_installer = PodTargetInstaller.new(sandbox, @project, pod_target, umbrella_headers_by_dir)
+              [pod_target.name, target_installer.install!]
+            end]
 
-            aggregate_targets.sort_by(&:name).each do |target|
-              target_installer = AggregateTargetInstaller.new(sandbox, project, target)
-              target_installer.install!
+            # Hook up system framework dependencies for the pod targets that were just installed.
+            pod_target_installation_result_values = pod_target_installation_results.values.compact
+            unless pod_target_installation_result_values.empty?
+              add_system_framework_dependencies(pod_target_installation_result_values)
             end
 
-            add_system_framework_dependencies
+            aggregate_target_installation_results = Hash[aggregate_targets.sort_by(&:name).map do |target|
+              target_installer = AggregateTargetInstaller.new(sandbox, @project, target)
+              [target.name, target_installer.install!]
+            end]
+
+            return [pod_target_installation_results, aggregate_target_installation_results]
```

Comment:
Could target installation results potentially be modelled as something clearer than a tuple array? - It's not super clear what is what without tracing code through to here what indexes are which.

Reply:
you are right, I didn't have a better way. Let me think about it and figure out if I can do so.

Reply:
@DanToml best I came up with is just create a `Struct` that has the two values, its definitely more readable now and avoids using `first` and `last`.

Please let me know what you think